netebpfext/net_ebpf_ext_bind.c

```diff
@@ -350,4 +371,89 @@ net_ebpf_ext_resource_release_classify(
     if (attached_client)
         net_ebpf_extension_hook_client_leave_rundown(attached_client);
     return;
-}
\ No newline at end of file
+}
+
+static uint64_t
+_ebpf_bind_context_create(
+    _In_reads_bytes_opt_(data_size_in) const uint8_t* data_in,
+    size_t data_size_in,
+    _In_reads_bytes_opt_(context_size_in) const uint8_t* context_in,
+    size_t context_size_in,
+    _Outptr_result_maybenull_ void** context)
+{
+    NTSTATUS status;
+    bind_md_t* bind_context = NULL;
+
+    if (!context) {
+        status = STATUS_INVALID_PARAMETER;
+        goto Exit;
+    }
+
+    *context = NULL;
+
+    if (context_in == NULL || context_size_in < sizeof(bind_md_t)) {
+        status = STATUS_INVALID_PARAMETER;
+        goto Exit;
+    }
+
+    bind_context =
+        (bind_md_t*)ExAllocatePoolUninitialized(NonPagedPool, sizeof(bind_md_t), NET_EBPF_EXTENSION_POOL_TAG);
+
+    if (!bind_context) {
+        status = STATUS_NO_MEMORY;
+        goto Exit;
+    }
+
+    // Copy the context from the caller.
+    memcpy(bind_context, context_in, sizeof(bind_md_t));
```

Comment:
Any reason why we aren't using ```RtlCopyMemory```? I understand that ```memcpy``` is probably a (hyper-optimized) compiler intrinsic, but for the longest time, usage of the 'Rtlxxx' variants was _strongly_ encouraged in kernel-mode components.  Was just curious if things have changed since.

Reply:
> 

<sigh> nevermind.  ```RtlCopyMemory``` and friends are #defined to point to their respective ```mem...``` equivalents (in ```wdm.h```, ```winnt.h```) :-D